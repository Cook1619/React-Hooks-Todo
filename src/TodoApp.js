import React, { useState } from "react";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import Appbar from "@material-ui/core/Appbar";
import Toolbar from "@material-ui/core/Toolbar";
import Grid from "@material-ui/core/Grid";
import TodoList from "./TodoList";
import TodoForm from "./TodoForm";

const TodoApp = () => {
  const intialTodos = [
    { id: 1, task: "Wash Car", completed: true },
    { id: 2, task: "Walk dog", completed: true },
    { id: 3, task: "Wash House", completed: false },
  ];
  const [todos, setTodos] = useState(intialTodos);
  const addTodo = (newTodo) => {
    setTodos([...todos, { id: 4, task: newTodo, completed: false }]);
  };
  return (
    <Paper
      style={{
        padding: 0,
        margin: 0,
        height: "100vh",
        backgroundColor: "#fafafa",
      }}
      elevation={0}
    >
      <Appbar color="primary" position="static" style={{ height: "64px" }}>
        <Toolbar>
          <Typography color="inherit">Todos with hooks</Typography>
        </Toolbar>
      </Appbar>
      <Grid container justify="center" style={{ marginTop: "2rem" }}>
        <Grid item xs={11} md={8} lg={4}>
          <TodoForm addTodo={addTodo} />
          <TodoList todos={todos} />
        </Grid>
      </Grid>
    </Paper>
  );
};

export default TodoApp;
