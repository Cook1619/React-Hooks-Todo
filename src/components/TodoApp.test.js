import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import TodoApp from "./TodoApp";

afterEach(cleanup);

test("it should take a snapshot", () => {
  const { asFragment } = render(<TodoApp />);
  expect(asFragment(<TodoApp />)).toMatchSnapshot();
});

describe("TodoApp", () => {
  test("renders the TodoApp component", () => {
    render(<TodoApp />);
    screen.getByText("LIST WITH HOOKS");
  });
});
