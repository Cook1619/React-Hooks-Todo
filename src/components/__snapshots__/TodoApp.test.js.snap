// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation0 MuiPaper-rounded"
    style="padding: 0px; margin: 0px; height: 100vh; background-color: rgb(250, 250, 250);"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorSecondary MuiPaper-elevation4"
      style="height: 64px;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit"
        >
          LIST WITH HOOKS
        </p>
      </div>
    </header>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      style="margin-top: 2rem;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-md-8 MuiGrid-grid-lg-4"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
          style="margin: 1rem 0px; padding: 0px 1rem;"
        >
          <form>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
              >
                Add New Todo
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
